### DISCUSSION TITLE
Infinity!

### DISCUSSION THREAD

**Discussion by adesutherland** [adesutherland/CREXX#401] [General]
Working on the decimal plugin and conversions / consistency with double precision - we have the following choice (one for the ARB?). This is about the string representation of non-numbers like infinity. On balance - I am tending to "inf" etc. because of C/Python convention but don't have a strong opinion either way. This is what I will use in the short term.

**C Standard Library (printf and strtod)**

The printf family of functions and strtod output and accept "inf" and "nan" by default in lowercase.
The output format can vary depending on the platform, but "inf" is typically used for infinity.
Uppercase versions ("INF" and "NAN") may be produced based on format specifiers (e.g., %F or %E).

**IEEE 754 Standard**

The IEEE 754 standard for floating-point arithmetic uses "Infinity" and "-Infinity" when formally describing special cases.
However, actual representations for infinity in many programming languages and libraries follow the "inf" convention.

**decNumber Library**

The decNumber library uses "Infinity" and "-Infinity" for representing decimal infinities in string form.
The string functions like decNumberToString typically produce "Infinity" rather than "inf".

**JSON and Other Data Formats**

In JSON, there is no standard representation for NaN or Infinity, but many implementations use "Infinity" and "-Infinity" as strings.

**Python and Other Languages**

Python represents positive and negative infinity as inf and -inf respectively.
When converting to strings, str(float('inf')) results in 'inf'.

**Comment by Peter-Jacob** [comment]
I am fine with inf/-inf/nan
